---
layout: page
title: 
permalink: /research/
---

My resume is located here: [PhuTVan-resume.pdf](PhuTVan-resume.pdf)

# A brief summary

I'm currently working as an analyst at [Fred Hutch](http://rglab.org), specializing in high-dimensional flow cytometry and RNASeq. I received my PhD at [Carnegie Mellon](https://cmu.edu/bio) where I designed and built a patented fluorescence imager to detect low-abundance proteins. Before grad school I worked on gene transcription networks and proteomics at the [Institute for Systems Biology](https://baliga.systemsbiology.net/). My greatest strengths are curiosity, adaptability and an exceptionally strong work ethic. 

# Major projects from most recent

### openCyto & ggCyto
As a postdoc, I helped extend [OpenCyto](http://opencyto.org), an open-source R software framework for gating high-dimensional flow-cytometry and mass-cytometry data. I also worked on [ggCyto](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6223365/), an R package that enables ggplot-style plotting of flow- and mass-cytometry datasets.
![ggCyto](/images/ggcyto-example.jpg "ggcyto-example.jpg")

### Structured Illumination Gel Imager (SIGI)
As part of my PhD thesis, I built a protein [imager](https://www.ncbi.nlm.nih.gov/pubmed/24935033) using structured illumination to increase dynamic range in order to detect rare protein in complex mixtures of cell extracts. This work was granted a US [patent](http://www.freepatentsonline.com/10362237.html) after I graduated.
![SIGI overview](/images/SIGI-operation.jpg "SIGI-operation.jpg")

### Archeal oxidative stress response network
Working at the Institute for Systems Biology as a research associate, I worked on a refined version of [Inferelator](https://www.ncbi.nlm.nih.gov/pubmed/16686963), an R algorithm for predicting genetic regulators of gene expression. This work enabled us to build and test a model of oxidative stress response in an extremophilic archaeon. This branch of the algorithm ended up being superseded, but you can see a version of code I worked on in this Github [repository](https://github.com/ptvan/inferelator-ancient).
![EGRIN_OS network](/images/EGRIN_OS-network.jpg "EGRIN_OS-network.jpg")

### Halobacterium PeptideAtlas 
As an intern working at ISB, I converted peptide mass spectra from vendor-specific binary formats into mzXML, mapped spectra to peptides then loaded the proteins into SQLServer-backed web [portal](https://peptideatlas.org). We also found biases in peptide detection depending on [the method used](https://www.ncbi.nlm.nih.gov/pmc/articles/PMC2643335/). 
![Halobacterium PeptideAtlas](/images/halopeptideatlas-peptidecount.jpg "halopeptideatlas-peptidecount.jpg")

### Nutria spread in Seattle
As University of Washington undergrads, my good friend Filip and I collected sightings of [nutria](https://en.wikipedia.org/wiki/Coypu) (Myocastor coypus) in western Washington, mainly in areas surrounding Union Bay. I ran a linear model to predict the spread of the species in Seattle using R and ArcGIS. This ended up informing a decision by the UW's Environmental Health and Safety not to undertake eradication efforts, since much of the population was not surviving through the winters anyway. 
![UBNA nutria map](/images/UBNA-model.jpg "UBNA-model.jpg")